libethash/data_sizes.h

```diff
@@ -33,213 +31,780 @@ extern "C" {
-        33491648U, 33601856U, 33712576U, 33822016U, 33932992U, 34042816U, 34153024U,
-        34263104U, 34373824U, 34485056U, 34594624U, 34704832U, 34816064U, 34926272U,
-        35036224U, 35146816U, 35255104U, 35367104U, 35478208U, 35588416U, 35698496U,
-        35808832U, 35918656U, 36029888U, 36139456U, 36250688U, 36360512U, 36471104U,
-        36581696U, 36691136U, 36802112U, 36912448U, 37022912U, 37132864U, 37242944U,
-        37354048U, 37464512U, 37574848U, 37684928U, 37794752U, 37904704U, 38015552U,
-        38125888U, 38236864U, 38345792U, 38457152U, 38567744U, 38678336U, 38787776U,
-        38897216U, 39009088U, 39117632U, 39230144U, 39340352U, 39450304U, 39560384U,
-        39671488U, 39781312U, 39891392U, 40002112U, 40112704U, 40223168U, 40332608U,
-        40443968U, 40553792U, 40664768U, 40774208U, 40884416U, 40993984U, 41105984U,
-        41215424U, 41326528U, 41436992U, 41546048U, 41655872U, 41768128U, 41878336U,
-        41988928U, 42098752U, 42209344U, 42319168U, 42429248U, 42540352U, 42649792U,
-        42761024U, 42871616U, 42981824U, 43092032U, 43201856U, 43312832U, 43423552U,
-        43533632U, 43643584U, 43753792U, 43864384U, 43974976U, 44084032U, 44195392U,
-        44306368U, 44415296U, 44526016U, 44637248U, 44746816U, 44858048U, 44967872U,
-        45078848U, 45188288U, 45299264U, 45409216U, 45518272U, 45630272U, 45740224U,
-        45850432U, 45960896U, 46069696U, 46182208U, 46292416U, 46402624U, 46512064U,
-        46623296U, 46733888U, 46843712U, 46953664U, 47065024U, 47175104U, 47285696U,
-        47395904U, 47506496U, 47615296U, 47726912U, 47837632U, 47947712U, 48055232U,
-        48168128U, 48277952U, 48387392U, 48499648U, 48609472U, 48720064U, 48830272U,
-        48940096U, 49050944U, 49160896U, 49271744U, 49381568U, 49492288U, 49602752U,
-        49712576U, 49822016U, 49934272U, 50042816U, 50154304U, 50264128U, 50374336U,
-        50484416U, 50596288U, 50706752U, 50816704U, 50927168U, 51035456U, 51146944U,
-        51258176U, 51366976U, 51477824U, 51589568U, 51699776U, 51809728U, 51920576U,
-        52030016U, 52140736U, 52251328U, 52361152U, 52470592U, 52582592U, 52691776U,
-        52803136U, 52912576U, 53020736U, 53132224U, 53242688U, 53354816U, 53465536U,
-        53575232U, 53685568U, 53796544U, 53906752U, 54016832U, 54126656U, 54236992U,
-        54347456U, 54457408U, 54569024U, 54679232U, 54789184U, 54899776U, 55008832U,
-        55119296U, 55231168U, 55341248U, 55451584U, 55562048U, 55672256U, 55782208U,
-        55893184U, 56002112U, 56113216U
+//         DataSetSizeBytesInit = 2^30,
+//         MixBytes = 128,
+//         DataSetGrowth = 2^23,
+//         HashBytes = 64,
+//         CacheMultiplier = 1024,
+//         j = 0},
+//     Reap[
+//       While[j < n,
+//        Module[{i = Floor[(DataSetSizeBytesInit + DataSetGrowth * j) / (CacheMultiplier * HashBytes)]},
+//         While[! PrimeQ[i], i--];
+//         Sow[i*HashBytes]; j++]]]][[2]][[1]]
+
+const size_t cache_sizes[2048] = {
```

Comment:
hard-coding this is utterly horrible. prefer to provide the function to calculate it alongside the data.